### ISSUE TITLE
How to add custom fields to the schema?

### ISSUE THREAD

**Issue by jkhaui** [wp-graphql/wp-graphql#1097]
Hey,

So I last used this plugin a year ago. I made a little extension that exposed a custom `user_meta` key and value pair from the WP database and added it to the schema. As far as I recall, the code worked when I wrote it, but it no longer works with this latest version of wp-graphql. I've traced the issue to the resolver. Here's the snippet in question:

```
use WPGraphQL\Types;

add_action( 'graphql_init', function() {
    add_filter( 'graphql_user_fields', function( $fields ) {

        $fields['userAvatarURL'] = [
            'type' => \WPGraphQL\Types::String(),
            'description' => __( 'Displays the user\'s profile picture URL.', 'gqlprofpic' ),
            'resolve' => function( $user ) {

                var_dump( $user );

                $wp_user_avatar = get_user_meta( $user->ID, 'wp_user_avatar', true );
                $wp_user_avatar = array_shift( $wp_user_avatar );

                return isset( $wp_user_avatar['avatar_url'] ) && ! empty( $wp_user_avatar['avatar_url'] ) ? $wp_user_avatar['avatar_url'] : null;

            },
        ];

        return $fields;

    } );
} );
```

When I check the network tab in Chrome, this is what's printed:
![graphql](https://misc-4.s3-ap-southeast-1.amazonaws.com/wp-graphql-example.png)

So in summary: I have a custom user_meta field in the database with the key 'wp_user_avatar', but it's not appearing in the resolver of my code. I've tried using https://docs.wpgraphql.com/extending/types/ as a guide, but I don't normally use PHP so it's difficult for me to follow. Any tips on how to fix this? 

Thanks

**Comment by jkhaui** [comment]
I've updated the code to make use of the register_fields API, but the same issue persists:

```
add_action( 'graphql_register_types', function() {
    register_graphql_field( 'User', 'userAvatarURL',[
        'type' => \WPGraphQL\Types::String(),
        'description' => __( 'Displays the user\'s profile picture URL.', 'gqlprofpic' ),
        'resolve' => function( $user ) {

            var_dump( $user );

            $wp_user_avatar = get_user_meta( $user->ID, 'wp_user_avatar', true );
            $wp_user_avatar = array_shift( $wp_user_avatar );

            return isset( $wp_user_avatar['avatar_url'] ) && ! empty( $wp_user_avatar['avatar_url'] ) ? $wp_user_avatar['avatar_url'] : null;

        },
    ] );
} );
```

**Comment by kidunot89** [comment]
Don't use `\WPGraphQL\Types` that namespace no longer contains the primitive types. Use `'String'` instead.

**Comment by jkhaui** [comment]
> `\WPGraphQL\Types` no longer contains the primitive types definitions. So don't use `\WPGraphQL\Types::string()`, instead use `'String'`.
> 
> If you was to use a type modifier like `list_of`, make an associative array with one entry. `array( 'list_of' => 'String' )`.

Great, thank you!

**Comment by jkhaui** [comment]
@kidunot89 the problem is that I can't access the user's ID property. It sounds simple but I haven't worked with PHP/WordPress for a while. Do you know how I should achieve this?

E.g. `var_dump( $user->{'data'}->{'data'}->{'ID'} );` doesn't work; it just returns null.

Thank you

**Comment by jasonbahl** [comment]
@jkhaui try `$user->id` instead of `$user->ID` as you are now passed an instance of the User Model instead of the WP_User.

**Comment by jkhaui** [comment]
@jasonbahl oh great, that does actually work for returning the User Model id. But in this case I believe I still need the WP_User ID (e.g. a number like 40, as opposed to a string like "dXNlcjo0MA==" returned by the User Model id) so I can use it with `get_user_meta`. Any ideas how I might achieve this?

**Comment by jasonbahl** [comment]
@jkhaui ah yes, sorry. That would be `$user->userId`

**Comment by jasonbahl** [comment]
Fwiw: https://github.com/wp-graphql/wp-graphql/blob/develop/src/Model/User.php#L177

**Comment by jkhaui** [comment]
Brilliant, thanks @jasonbahl !

**Comment by benregan** [comment]
Holy Moses.... this just saved me... I've been banging my head on this for 3 days. Thank you so much!
